### PR TITLE
Test and process result only for latest, upstream and 0.2.1

### PR DESCRIPTION
**Pull request by ncclementi**
- Closes https://github.com/coiled/coiled-runtime/issues/586

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -274,7 +266,7 @@ jobs:
           environment-file: ci/environment-dashboard.yml
 
       - name: Generate dashboards
-        run: python dashboard.py -d benchmark.db -o static -b coiled-latest-py3.9 coiled-0.1.0-py3.9
+        run: python dashboard.py -d benchmark.db -o static -b coiled-latest-py3.9 coiled-0.2.1-py3.9
```
Comment (ncclementi) [inline .github/workflows/tests.yml:269]: I think this change is needed, but I can be wrong, 
@crusaderky please correct me if I'm wrong.

**Comment by ncclementi** [comment]
In the PR, the artifact, only builds this, 

Screen Shot 2022-12-07 at 10 45 18 AM


but I do not know if on main, because we have the previous deploys it'll keep those too. Does anyone know how the deployment works?

**Comment by ncclementi** [comment]
The error are due to a known cause see https://github.com/coiled/coiled-runtime/issues/521

File: .github/workflows/tests.yml (hunk 1 above)
Comment (crusaderky) [reply .github/workflows/tests.yml:269]: Yes. The intent is to show the differences of `latest` and `upstream` vs. the latest published version.

File: dashboard.py [hunk 2]
```diff
@@ -716,7 +716,8 @@ def main() -> None:
     engine = sqlalchemy.create_engine(f"sqlite:///{args.db_file}")
     df = pandas.read_sql(
         "select * from test_run where platform = 'linux' "
-        "and call_outcome in ('passed', 'failed')",
+        "and call_outcome in ('passed', 'failed')"
+        "and coiled_runtime_version >= '0.2.1' ",
```
[the remark this answers is not in the document]
Comment (crusaderky) [reply dashboard.py:720]: I don't think I like this.
If we don't care about 0.0.4 and 0.1.0 anymore, shouldn't we wipe the database of them with an alembic rule instead?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (crusaderky) [inline .github/workflows/tests.yml:269]:
```suggestion
        # Show differences of upstream vs. latest unpublished vs. latest publshed
        run: python dashboard.py -d benchmark.db -o static -b coiled-latest-py3.9 coiled-0.2.1-py3.9
```

File: dashboard.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ncclementi) [reply dashboard.py:720]: My understanding is that we do not want to run them anymore, but we would probably want to keep the history. That's why I was not touching the database but the plotting part only.

**Review by crusaderky — APPROVED** [review]

**Pull request merged**